test/framework/src/main/java/org/elasticsearch/test/CorruptionUtils.java

```diff
@@ -78,8 +78,19 @@ public static void corruptFile(Random random, Path... files) throws IOException
                 checksumBeforeCorruption = CodecUtil.retrieveChecksum(input);
             }
             try (FileChannel raf = FileChannel.open(fileToCorrupt, StandardOpenOption.READ, StandardOpenOption.WRITE)) {
+                long maxPosition = raf.size();
+
+                if (fileToCorrupt.getFileName().toString().endsWith(".cfs") && maxPosition > Integer.BYTES) {
+                    // TODO: it is known that Lucene does not check the tail of CFS file (CompoundFileS, like an archive), see note at
+                    // https://github.com/apache/lucene-solr/blob/branch_7_5/lucene/core/src/java/org/apache/lucene/codecs/lucene50/
+                    // Lucene50CompoundReader.java#L82
+                    // so far, don't corrupt checksum (last 4 bytes) of cfs file
+                    maxPosition -= Integer.BYTES;
```

Comment:
This is a little confusing: the checksum is actually the last 8 bytes of the file, but the first 4 of those should be 0 (and we verify that). Could (a) the comment be more detailed and (b) the `Integer.BYTES` just say `4` to avoid this confusion?